### PR TITLE
gh-119180: Add evaluate functions for type params and type aliases

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Implements the following section of PEP 749: https://peps.python.org/pep-0749/#deferred-evaluation-of-pep-695-and-696-objects

* Issue: gh-119180

### CODE REVIEW

File: Objects/genericaliasobject.c [hunk 1]
```diff
@@ -51,70 +52,7 @@ ga_traverse(PyObject *self, visitproc visit, void *arg)
...
-
-done:
-    Py_XDECREF(qualname);
-    Py_XDECREF(module);
-    return rc;
-}
-
-static int
-ga_repr_items_list(PyUnicodeWriter *writer, PyObject *p)
+ga_repr_items_list(_PyUnicodeWriter *writer, PyObject *p)
```
Comment (carljm) [inline Objects/genericaliasobject.c]: It looks like this change from `PyUnicodeWriter *` to `_PyUnicodeWriter *` in this signature causes a lot of compiler warnings below.

**Review by carljm — APPROVED** [review]

File: Objects/genericaliasobject.c (hunk 1 above)
Comment (JelleZijlstra) [reply Objects/genericaliasobject.c]: Thanks, there was a change to this API since I first wrote the code and I may not have merged properly.

**Comment by JelleZijlstra** [comment]
Going to merge this if the refleak buildbots are happy

**Comment by sobolevn** [comment]
I submitted https://github.com/python/cpython/pull/122362

**Pull request merged**